### ISSUE TITLE
nodemailer no longer working with Node 0.10.xx?

### ISSUE THREAD

**Issue by SebastianMetzger** [andris9/Nodemailer#533]
First of all, thank you for creating and maintaining this project. It served us well in the last months!

We observed that no more Emails were sent from our server last week after doing a new release which included a fresh 'npm install'.
OS: Ubuntu Server
Node version: 0.10.25
Transport: SMTP

There is no direct error after sending an email, the connection rather times out after a very long time with an 'ETIMEDOUT' error.

I tinkered around with this a bit on my local windows machine and found out, that the e-mails are sent out with Node 4.2.2, but not with Node 0.10.xx.

I saw some recent commits in the dependency project such as 'smtp-connection'. Can this have something to do with our issue?

**Comment by SebastianMetzger** [comment]
UPDATE: I manually reverted the 'smtp-connection' to 1.3.1 and everything works fine again.

**Comment by andris9** [comment]
Are you using some special configuration? If I try to send mail with latest node v0.10 then everything seems to work as intended

**Comment by SebastianMetzger** [comment]
```javascript
'use strict';
var nodemailer = require('nodemailer');

var smtp = nodemailer.createTransport({
  host: 'our.email.host',
  port: 587,
  secureConnection: false,
  auth: {
    user: 'mail@ourdomain.com',
    pass: '12341234'
  }
});

smtp.sendMail({
  from: 'mail@ourdomain.com',
  to: ['s.metzger@ourdomain.com'], // An array if you have multiple recipients.            
  subject: 'Test',
  text: 'Test 123 123 123'
}, function(err, info) {
  if (err) {
    console.log('ERROR SENDING MAIL');
    console.log(err);
  } else {
    console.log('EMAIL SENDING SUCCESS');
    console.log(info);
  }
});
```
Not really, above is the code of my test case.

It only works when I manually fix the dependency inside the nodemailer-smtp-transport module of smtp-connection to 1.3.1.

**Comment by andris9** [comment]
Which version of smtp-connection did you downgrade? Have you tried to use the latest, v1.3.6?

**Comment by SebastianMetzger** [comment]
Yes, it did not work with v1.3.6
There is no error, the application just hangs and eventually times out. (after several minutes)

Might have to do with some mail server config?

**Comment by andris9** [comment]
The diff between 1.3.1 and 1.3.6 isn't that big ([see here](https://github.com/andris9/smtp-connection/compare/v1.3.1...v1.3.6?w=1)). I haven't been able to reproduce it myself (I tried against Gmail on port 587 where it worked fine with node v0.10.25), so it would be interesting to know what is causing the issue for you.

**Comment by selcukfatihsevinc** [comment]
I have the same problem. Both nodemailer and and nodemailer-smtp-pool stopped working. When i downgraded to smtp-connection 1.3.1, it works.

**Comment by andris9** [comment]
Found the issue and published fixed version as v1.3.7. There's apparently a 'finish' event set for the socket, so when `socket.removeAllListeners()` was called for the plaintext socket when upgrading to TLS, the 'finish' event was also cleared and upgrade never finished. https://github.com/andris9/smtp-connection/commit/8fa722a3b207b6ea12484f900b206c658c388762
